### PR TITLE
[tensor_trainer] Determined g_cond_wait by train_complete when receive EOS

### PR DESCRIPTION
**Pull request by songgot**
Determined g_cond_wait by train_complete when receive EOS
 - To avoid dead lock, g_cond_wait is determined by train_complete value of GstTensorTrainerFrameworkInfo when receive EOS 

**Self evaluation:**
1. Build test: [*]Passed [ ]Failed []Skipped
2. Run test: [*]Passed [ ]Failed []Skipped

### CODE REVIEW

**Comment by songgot** [comment]
Thank you.

File: gst/nnstreamer/elements/gsttensor_trainer.c [hunk 1]
```diff
@@ -541,7 +541,14 @@ gst_tensor_trainer_chain (GstPad * sinkpad, GstObject * parent,
   if (trainer->total_invoke_num == 1
       || trainer->total_invoke_num ==
       trainer->prop.num_train_samples + trainer->prop.num_valid_samples) {
-
+#if 0
```
Comment (jaeyun-jung) [inline gst/nnstreamer/elements/gsttensor_trainer.c:544]: plz remove these block.

Comment (songgot) [reply gst/nnstreamer/elements/gsttensor_trainer.c:544]: okay, thank you.

Comment (songgot) [reply gst/nnstreamer/elements/gsttensor_trainer.c]: Done ([c9511e6](https://github.com/nnstreamer/nnstreamer/pull/4023/commits/c9511e6d1a5b12ae09ffa85ac5598e8457bc01fa))
Thank you.

File: gst/nnstreamer/elements/gsttensor_trainer.c [hunk 2]
```diff
@@ -703,19 +703,27 @@ gst_tensor_trainer_sink_event (GstPad * sinkpad, GstObject * parent,
...
 
   switch (GST_EVENT_TYPE (event)) {
     case GST_EVENT_EOS:
-      GST_INFO_OBJECT (trainer, "get GST_EVENT_EOS event..state is %d",
-          GST_STATE (trainer));
-      g_mutex_lock (&trainer->trainer_lock);
-      GST_INFO_OBJECT (trainer, "wait for training_complete_cond signal");
-      g_cond_wait (&trainer->training_complete_cond, &trainer->trainer_lock);
-      g_mutex_unlock (&trainer->trainer_lock);
+      GST_ERROR ("trainer->privateData: %p", trainer->privateData);
```
Comment (jaeyun-jung) [inline gst/nnstreamer/elements/gsttensor_trainer.c:714]: Recommend not printing priv-data (error? to print address, IMO it is unnecessary)

File: gst/nnstreamer/elements/gsttensor_trainer.c [hunk 3]
```diff
@@ -703,19 +703,27 @@ gst_tensor_trainer_sink_event (GstPad * sinkpad, GstObject * parent,
...
+      trainer->fw->getFrameworkInfo (trainer->fw, NULL, trainer->privateData,
+          &info);
+      GST_INFO_OBJECT (trainer, "training_complete is %d",
+          info.is_training_complete);
+      if (!info.is_training_complete) {
+        GST_INFO_OBJECT (trainer, "get GST_EVENT_EOS event..state is %d",
```
Comment (jaeyun-jung) [inline gst/nnstreamer/elements/gsttensor_trainer.c:720]: Suggestion to reduce log msg, w/line 717, got GST_EVENT_EOS event but training is not completed, state is %d.

**Review by gichan-jang — APPROVED** [review]
LGTM,
It would be nice to modify the messages more meaningful.

File: gst/nnstreamer/elements/gsttensor_trainer.c (hunk 3 above)
Comment (songgot) [reply gst/nnstreamer/elements/gsttensor_trainer.c:720]: okay, thank you!!

**Comment by songgot** [comment]
Thank you, I will do that.

File: gst/nnstreamer/elements/gsttensor_trainer.c (hunk 2 above)
Comment (songgot) [reply gst/nnstreamer/elements/gsttensor_trainer.c]: okay, thank you.

**Review by jaeyun-jung — APPROVED** [review]

**Pull request merged**